src/rs.cpp

```diff
@@ -557,6 +557,21 @@ rs2_stream_profile* rs2_clone_video_stream_profile(const rs2_stream_profile* mod
 }
 HANDLE_EXCEPTIONS_AND_RETURN(nullptr, mode, stream, index, format, width, height, intr)
 
+const rs2_raw_data_buffer* rs2_build_raw_data(rs2_device* device, unsigned opcode, unsigned param1, unsigned param2,
+    unsigned param3, unsigned param4, void* data, unsigned size_of_data, rs2_error** error) BEGIN_API_CALL
+{
+    VALIDATE_NOT_NULL(device);
+
+    auto debug_interface = VALIDATE_INTERFACE(device->device, librealsense::debug_interface);
+
+    auto raw_data_buffer = static_cast<uint8_t*>(data);
+    std::vector<uint8_t> buffer_to_send(raw_data_buffer, raw_data_buffer + size_of_data);
```

Comment:
`raaw_data_buffer` may bel `nullptr`, no?

Reply:
Also: this essentially copies the data, which I'd like to avoid.
Any way to not copy?

Reply:
do you think that "rs2_send_and_receive_raw_data" should be fixed too ?

Reply:
If we don't break backwards compatibility, yes